lib/string.gi

```diff
@@ -1258,9 +1258,121 @@ InstallGlobalFunction(StringOfMemoryAmount, function(m)
     Append(s, units[shift+1]);
     return s;
 end);
-
-        
     
+InstallGlobalFunction(PrintToFormatted, function(stream, s, args...)
+    local pos, len, outstr, nextbrace, endbrace,
+          argcounter, var,
+          splitBracket, toprint;
+
+    splitBracket := function(string, startpos, endpos)
+      local posbang, type;
+      posbang := Position(s, '!', startpos-1);
+#      Print(startpos, ":",endpos, ":",posbang,"\n");
+      if posbang = fail or posbang > endpos then
+        posbang := endpos + 1;
+      fi;
+      type := string{[posbang + 1 .. endpos]};
```

Comment:
In the documentation, you call this the "format", and now the "type". I'd recommend to use uniform language, to make it easier to match documentation and code.